woocommerce/woocommerce-android
# Variable bundles creation

Part of: #9541

### Why
As part of the support for extensions milestone 2, we are working on support bundle products in the order creation flow.

### Description
This PR includes the necessary changes to display, validate, and create variable products. With the changes introduced here, it is now possible to create products containing variable products, select the variations of these variable products, and validate issues related to variable products in the bundle configuration.


### Testing instructions

#### Prerequisites 

1.  Install the product bundles extension
2. Create a bundle product that includes a variable product

#### Testing

1. Open the orders tab
2. Tap on create a new order (+)
3. Select the product bundle containing the variable product
4. Check that the configuration displays the variable product
5. Check that pressing on the variable product lets you select the variation
6. Check that an error message is displayed in the UI when the variable product quantity is greater than 0 and no variation is selected.
7. Select the bundled product containing a variable product.
8. Create the bundled products containing a variable product.
9. Check that the bundled product is created successfully

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/ade0f4f2-b8aa-45ac-a145-f7c74feef4ed
